mantle/cmd/kola/devshell.go

```diff
@@ -215,6 +217,15 @@ loop:
 	// Later Ctrl-c after this should just kill us
 	signal.Reset(os.Interrupt)
 
+	// This one takes ownership of the readyReader
```

Comment:
```suggestion
```

I understand the influence of Rust here, but the comment is incorrect.

Reply:
Hmm...the way I see it the flow is:

- create readyReader in main goroutine
- spawn goroutine which reads readyReader and writes to readychan when it finds a specific message, then *it exits*
- main goroutine now holds sole reference to readyReader
- swawn goroutine which reads readyReader watching for poweroff
- after that, main goroutine no longer references readyReader, so we can think of the second goroutine as the "owner"

Does that make sense?

Reply:
Hmm OK perhaps "owner" in the Rust sense is indeed wrong here it's more like "active &mut reference"?  Is that what you mean?

Reply:
(I squashed in your change)

>  The last reference of readyReader is still in scope to func runDevShellSSH(...) albeit its only used in the go routine in question.

Hmm yes.  Though now I'm curious whether the Go GC uses more precise rooting based on scope.  The compiler knows the variable isn't referenced after that, so presumbly it could know not to scan the local roots after execution reaches that point.  But maybe that's more code/data, so it may be more conservative indeed and just have "roots potentially in scope for the whole function".